src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,213 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactElement, useCallback, useMemo, useState } from 'react'
+import { Modal as AntModal } from 'antd'
+import classNames from 'classnames'
+
+import { Shape, Type } from '../Button/Button.types'
+import { BodyM } from '../Typography/BodyX/BodyM'
+import { Button } from '../Button'
+import { H4 } from '../Typography/HX/H4'
+import { Icon } from '../Icon'
+import { ModalProps } from './Modal.props'
+import { Size } from './Modal.types'
+import styles from './Modal.module.css'
+import { useTheme } from '../../hooks/useTheme'
+
+const { Circle } = Shape
+const { Ghost } = Type
+const { Small, Large, FullScreen } = Size
+const {
+  asideLabel,
+  asideLabelWrapper,
+  body,
+  bodyFullWidth,
+  content,
+  contentWrapper,
+  footerButtons,
+  modal,
+  modalSm,
+  modalLg,
+  modalFs,
+} = styles
+
+/**
+ * Modal dialog that opens in overlay and allows to perform flows of actions, or to show specific
+ * but less relevant information than those displayed on the underlying page.
+ *
+ * @link https://ant.design/components/modal
+ * @returns {Modal} Modal component
+ */
+export const Modal = ({
+  aside,
+  children,
+  docLink,
+  footer,
+  isBodyFullWidth,
+  isClosable,
+  isVisible,
+  onCloseClick,
+  size,
+  title,
+}: ModalProps): ReactElement => {
+  const { palette } = useTheme()
+
+  const docLinkIcon = useMemo(() => (
+    <Icon color={palette?.action?.link?.active} name="PiBookOpen" size={16} />
+  ), [palette?.action?.link?.active])
+
+  const onClickDocLink = useCallback(() => window.open(docLink, '_blank'), [docLink])
+
+  const modalClassNames = useMemo(() => classNames([
+    modal,
+    size === Small && modalSm,
+    size === Large && modalLg,
+    size === FullScreen && modalFs,
+  ]), [size])
+  const modalBodyClassNames = useMemo(() => classNames([
+    body,
+    isBodyFullWidth && bodyFullWidth,
+  ]), [isBodyFullWidth])
+  const modalContentWrapperClassNames = useMemo(() => classNames([
+    content,
+    size !== Small && aside && contentWrapper,
+  ]), [aside, size])
+
+  const [isAsideOpen, setIsAsideOpen] = useState(false)
+
+  const modalTitle = useMemo(() => (
+    <div className={styles.title}>
+      {title && <>
+        <H4 ellipsis={{ rows: 1, tooltip: title }}>{title}</H4>
+        {docLink && <div className={styles.docLink}>
+          <Button
+            icon={docLinkIcon}
+            shape={Circle}
+            type={Ghost}
+            onClick={onClickDocLink}
+          />
+        </div>}
+      </>}
+    </div>
+  ), [docLink, docLinkIcon, onClickDocLink, title])
+
+  const modalFooter = useMemo(() => {
+    const { buttons, extra } = footer || {}
+    return (
+      <div className={styles.footer}>
+        {footer && <>
+          <div className={footerButtons}>
+            {buttons}
+          </div>
+          {extra}
+        </>}
+      </div>
+    )
+  }, [footer])
+
+  const modalContent = useMemo(() => {
+    if (size !== Small && aside) {
```

Comment:
I think it is easier to understand if we invert the condition:

```suggestion
    if (size == Small || !aside) {
      return (
      <div className={content}>
        {children}
      </div>
    )}
```

Reply:
I converted this condition to a const called `hasAside` which I put at the top of the file to facilitate the assignment of `classNames`. In this new scenario it seems more useful to me to keep it as is (otherwise I would have to call the const `doesNotHaveAside`)